src/main/java/com/gradle/CommonCustomUserDataGradlePlugin.java

```diff
@@ -43,38 +43,47 @@ public void apply(Object target) {
         }
     }
 
-    private void applySettingsPlugin(Settings settings) {
-        settings.getPluginManager().withPlugin("com.gradle.enterprise", __ -> {
-            CustomGradleEnterpriseConfig customGradleEnterpriseConfig = new CustomGradleEnterpriseConfig();
-
-            Object extension = settings.getExtensions().getByName("gradleEnterprise");
-            GradleEnterpriseExtension gradleEnterprise = ProxyFactory.createProxy(extension, GradleEnterpriseExtension.class);
-            customGradleEnterpriseConfig.configureGradleEnterprise(gradleEnterprise);
-
-            BuildScanExtension buildScan = gradleEnterprise.getBuildScan();
-            customGradleEnterpriseConfig.configureBuildScanPublishing(buildScan);
-            CustomBuildScanEnhancements buildScanEnhancements = new CustomBuildScanEnhancements(buildScan, providers, settings.getGradle());
-            buildScanEnhancements.apply();
-
-            BuildCacheConfiguration buildCache = settings.getBuildCache();
-            customGradleEnterpriseConfig.configureBuildCache(buildCache);
+    public static void apply(Object gradleEnterprise, ProviderFactory providers, Settings settings) {
+        applySettingsPlugin(
+                gradleEnterprise,
+                providers,
+                settings
+        );
+    }
 
-            // configuration changes applied in this block will override earlier configuration settings,
-            // including those set in the settings.gradle(.kts)
-            Action<Settings> settingsAction = ___ -> {
-                Overrides overrides = new Overrides(providers);
-                overrides.configureGradleEnterprise(gradleEnterprise);
-                overrides.configureBuildCache(buildCache);
-            };
+    private static void applySettingsPlugin(Object gradleEnterpriseExtension, ProviderFactory providers, Settings settings) {
```

Comment:
@facewindu your indentation and spacing settings are different than that of this repository. can you make them consistent in order to slim up the diff?
it's hard to see that there are only changes to a few lines.

Reply:
Most of the diff you see comes from the fact that the `settings.getPluginManager().withPlugin()` callback has been extracted out. Therefore there is one less indentation level.